simpleclient/src/main/java/io/prometheus/client/stream/CKMSStream.java

```diff
@@ -0,0 +1,331 @@
+package io.prometheus.client.stream;
+
+/**
+ * Created by warebot on 12/27/15.
+ */
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+public class CKMSStream<T extends Number & Comparable<T>> implements Stream<T> {
```

Comment:
Like we talked - this class needs some documentation and if available a reference to an article